kv/task.go

```diff
@@ -1694,3 +1694,134 @@ func taskRunKey(taskID, runID influxdb.ID) ([]byte, error) {
 
 	return []byte(string(encodedID) + "/" + string(encodedRunID)), nil
 }
+
+func (s *Service) taskOwnerIDUpMigration(ctx context.Context, store Store) error {
+	var ownerlessTasks []*influxdb.Task
+	// loop through the tasks and collect a set of tasks that are missing the owner id.
+	err := store.View(ctx, func(tx Tx) error {
+		taskBucket, err := tx.Bucket(taskBucket)
+		if err != nil {
+			return influxdb.ErrUnexpectedTaskBucketErr(err)
+		}
+
+		c, err := taskBucket.ForwardCursor([]byte{})
+		if err != nil {
+			return influxdb.ErrUnexpectedTaskBucketErr(err)
+		}
+
+		// free cursor resources
+		defer c.Close()
+
+		for k, v := c.Next(); k != nil; k, v = c.Next() {
+			kvTask := &kvTask{}
+			if err := json.Unmarshal(v, kvTask); err != nil {
+				return influxdb.ErrInternalTaskServiceError(err)
+			}
+
+			t := kvToInfluxTask(kvTask)
+
+			if !t.OwnerID.Valid() {
+				ownerlessTasks = append(ownerlessTasks, t)
+			}
+		}
+		return c.Err()
```

Comment:
```suggestion
if err := c.Err(); err != nil {
    return err
}

return c.Close()
```

Reply:
Oh I see the defer close up the top. I guess you could do that too.

Reply:
I keep forgetting our close has an error. I will commit your suggestion and remove the defer.